src/fft_fftw3_f03.f90

```diff
@@ -1400,13 +1434,15 @@ subroutine fft_3d_c2r(in_c, out_r)
             else
                call transpose_x_to_y(wk1, wk2_r2c, sp)
             end if
-            call c2c_1m_y(wk2_r2c, plan(2, 2))
+            if (.not.(skip_y_c2c)) call c2c_1m_y(wk2_r2c, plan(2, 2)) !bm_issue337
          else  ! in_c==wk2_r2c if 1D decomposition
-            if (inplace) then
-               call c2c_1m_y(in_c, plan(2, 2))
-            else
-               call c2c_1m_y(wk1, plan(2, 2))
-            end if
+            if (.not.(skip_y_c2c)) then !bm_issue337
```

Comment:
It might be easier to read and more consistent with the previous lines to have
```
            if (inplace) then
               if (.not.(skip_y_c2c)) call c2c_1m_y(in_c, plan(2, 2))
            else
               if (.not.(skip_y_c2c)) call c2c_1m_y(wk1, plan(2, 2))
            end if
```

Reply:
being pedantic, wouldn't you put the skip test outside?
```
if (.not. skip_y_c2c) then
  if (inplace) then
    call ...
  else
    call ...
  end if
end if
```

Reply:
The question no longer holds if we skip inside the subroutine `c2c_1m_*`